playground/ssr/src/circular-import/b.js

```diff
@@ -0,0 +1,5 @@
+export function getB() {
+  return '__B__'
+}
+
+export { A } from './a'
```

Comment:
```suggestion
export { A } from './a'

export function getB() {
  return '__B__'
}
```
If I change this file like above, the test fails with `getB is not a function`. But this still works if I run it with `node ./playground/ssr/src/circular-import/index.js` (the `.js` extension needs to be appended to imports). 

I guess while this PR fixes in some cases, it breaks in some cases.

I don't know a way to fix this completely but I feel I definitely need to read https://262.ecma-international.org/14.0/#sec-example-cyclic-module-record-graphs

related: https://github.com/vitejs/vite/issues/14048#issuecomment-1672668515

Reply:
Good catch. Yes, It's a little tricky to fix completely. I'm guessing it would need to be in two phases as you said in https://github.com/vitejs/vite/issues/14048#issuecomment-1672668515

For this PR, I think the current import/export order is more intuitive.

Reply:
I've discussed this with Sapphi. While we'd like this fix too, the current PR does improves some areas with circular imports and doesn't regress this case (`getB is not a function` would also happen in `main`). So we can defer this to later and get this PR merged for now.